enso/sample/__init__.py

```diff
@@ -0,0 +1,53 @@
+import numpy as np
+import logging
+from sklearn.metrics.pairwise import pairwise_distances
+
+
+def sample(sampler, data, train_indices, train_size):
+    sampler = Sampler._class_for(sampler)(data, train_indices, train_size)
+    return sampler.sample()
+
+
+class Sampler(object):
+    """Base class for all samples."""
+
+    def __init__(self, data, train_indices, train_size):
+        self.data = data
+        self.train_indices = train_indices
+        self.train_size = train_size
+
+    def sample(self):
+        raise NotImplementedError
+
+    @classmethod
+    def _class_for(cls, sampler_string):
+        for subclass in cls.__subclasses__():
+            if subclass.__name__ == sampler_string:
+                return subclass
+        raise ValueError("Invalid sampler attempted ({})".format(sampler_string))
+
+
+class Random(Sampler):
+    def sample(self):
+        return np.random.choice(self.train_indices, self.train_size, replace=False)
+
+
+class KCenter(Sampler):
+    @property
+    def points(self):
+        return np.array(point) for point in self.data]
+
+    def sample(self):
+        random_center = np.random.choice(self.train_indices)
+        centers = [random_center]
+        while len(centers) < self.train_size:
+            mins = np.min(self.distances[centers], axis=0)
+            center = np.argmax(mins)
+            centers.append(center)
+        return centers
+    
+    @property
+    def distances(self):
```

Comment:
Nice clean implementation. There's maybe a higher level question about if we want to support something like sharing the distance computations across samplers, but out of scope for this pr

Reply:
yes, I assume we might want to abstract this but figured i'd wait until we have another sampler to design the right abstraction.